tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/AspNetWebApi2Integration.cs

```diff
@@ -96,45 +96,58 @@ internal static void UpdateSpan(IHttpControllerContext controllerContext, Span s
                 {
                 }
 
+                IDictionary<string, object> routeValues = null;
+                try
+                {
+                    routeValues = controllerContext.RouteData.Values;
+                }
+                catch
+                {
+                }
+
                 string resourceName;
 
-                if (route != null)
+                string controller = string.Empty;
+                string action = string.Empty;
+                string area = string.Empty;
+
+                if (route != null && routeValues is not null)
```

Comment:
You're just trolling me by mixing `!= null` and `is not null` in the same line, right? 😉

Reply:
Gotta keep you on your toes 😛

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2517/commits/9c4d6538bd17293c31c3187e07edfdbcfc916ff4 😉